lib/app_registry.js

```diff
@@ -2,45 +2,55 @@
 'use strict'
 
 import React from 'react'
-import { AppRegistry, Text } from 'react-native'
-import { RootContainer } from 'react-relay'
+import { AppRegistry } from 'react-native'
+import Relay from 'react-relay'
 
 import Spinner from './components/spinner'
+import LoadFailureView from './components/load_failure_view'
 import Containers from './containers'
 import Routes from './relay/routes'
 
-class Artist extends React.Component {
-  render() {
-    const artistRoute:Artist = new Routes.Artist({ artistID: this.props.artistID })
+class RootContainer extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = { retrying: false }
+  }
 
+  render() {
     // https://facebook.github.io/relay/docs/guides-root-container.html
-    return (<RootContainer
-             Component={Containers.Artist}
-             route={artistRoute}
+    return (<Relay.RootContainer
```

Comment:
I assume you'd want this on the Home too? Perhaps the re-try work could go into it's own artsy-relay-root container?